### PR TITLE
Do not generate dummy exports for PS on GFX10

### PR DESCRIPTION
**Pull request by perlfu**
LLVM backend is now aware of color/depth export information and
will not generate a null export for kills if there are no exports
in the shader.

This enables no export pixel shaders on GFX10.

### CODE REVIEW

**Comment by amdvlk-admin** [comment]
### Test summary for commit 9c922a7d9f0ec410e280c47316af309fa6d8f090
<details><summary>Driver commits used in build</summary><ul>

<li>CWPACK: amd-master 7387247eb9889ddcabbc1053b9c2052e253b088e</li>
<li>METROHASH: amd-master 3c566dd9cda44ca7fd97659e0b53ac953f9037d2</li>
<li>PAL: dev 408fbcdd1249dea3469bbd3426c29ca344386d6f</li>
<li>SPVGEN: dev bb3da8bc00bcf4373d563bf9b3c74a80a8b28fe6</li>
<li>XGL: dev 5b352af27e8877ab65b95636d4bed34ebbe02115</li>
<li>LLVM-PROJECT: amd-gfx-gpuopen-dev 35321a037d4e75d01fbf6b2264acb98ba4f72563</li>
</ul></details>

<details><summary>CTS tests (Failed: 0/228032)</summary><ul><li>Built with version 1.2.5.2</li>
<details open><summary>Rhel 8.2, Gfx10</summary><ul>

<li>Passed: 32108/57008 (56.3%)</li>
<li>Failed: 0/57008 (0.0%)</li>
<li>Not Supported: 24900/57008 (43.7%)</li>
</ul></details>

<details open><summary>Ubuntu 18.04, Gfx9</summary><ul>

<li>Passed: 32108/57008 (56.3%)</li>
<li>Failed: 0/57008 (0.0%)</li>
<li>Not Supported: 24900/57008 (43.7%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx8</summary><ul>

<li>Passed: 32488/57008 (57.0%)</li>
<li>Failed: 0/57008 (0.0%)</li>
<li>Not Supported: 24520/57008 (43.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx103</summary><ul>

<li>Passed: 35419/57008 (62.1%)</li>
<li>Failed: 0/57008 (0.0%)</li>
<li>Not Supported: 21589/57008 (37.9%)</li>
</ul></details>


</ul></details>

**Comment by amdvlk-admin** [comment]
### Test summary for commit 170cb29da6cc614a7460b2ea72edcdd96503e5d2
<details><summary>Driver commits used in build</summary><ul>

<li>CWPACK: amd-master 7387247eb9889ddcabbc1053b9c2052e253b088e</li>
<li>METROHASH: amd-master 3c566dd9cda44ca7fd97659e0b53ac953f9037d2</li>
<li>PAL: dev 408fbcdd1249dea3469bbd3426c29ca344386d6f</li>
<li>SPVGEN: dev bb3da8bc00bcf4373d563bf9b3c74a80a8b28fe6</li>
<li>XGL: dev 5b352af27e8877ab65b95636d4bed34ebbe02115</li>
<li>LLVM-PROJECT: amd-gfx-gpuopen-dev 35321a037d4e75d01fbf6b2264acb98ba4f72563</li>
</ul></details>

<details><summary>CTS tests (Failed: 0/228032)</summary><ul><li>Built with version 1.2.5.2</li>
<details open><summary>Rhel 8.2, Gfx10</summary><ul>

<li>Passed: 32108/57008 (56.3%)</li>
<li>Failed: 0/57008 (0.0%)</li>
<li>Not Supported: 24900/57008 (43.7%)</li>
</ul></details>

<details open><summary>Ubuntu 18.04, Gfx9</summary><ul>

<li>Passed: 32108/57008 (56.3%)</li>
<li>Failed: 0/57008 (0.0%)</li>
<li>Not Supported: 24900/57008 (43.7%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx8</summary><ul>

<li>Passed: 32488/57008 (57.0%)</li>
<li>Failed: 0/57008 (0.0%)</li>
<li>Not Supported: 24520/57008 (43.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx103</summary><ul>

<li>Passed: 35419/57008 (62.1%)</li>
<li>Failed: 0/57008 (0.0%)</li>
<li>Not Supported: 21589/57008 (37.9%)</li>
</ul></details>


</ul></details>

**Comment by amdvlk-admin** [comment]
### Test summary for commit 4f8d0ac99ef90ae6f9b8363c49636c16882618d7
<details><summary>Driver commits used in build</summary><ul>

<li>CWPACK: amd-master 7387247eb9889ddcabbc1053b9c2052e253b088e</li>
<li>METROHASH: amd-master 3c566dd9cda44ca7fd97659e0b53ac953f9037d2</li>
<li>PAL: dev 408fbcdd1249dea3469bbd3426c29ca344386d6f</li>
<li>SPVGEN: dev bb3da8bc00bcf4373d563bf9b3c74a80a8b28fe6</li>
<li>XGL: dev 5b352af27e8877ab65b95636d4bed34ebbe02115</li>
<li>LLVM-PROJECT: amd-gfx-gpuopen-dev 35321a037d4e75d01fbf6b2264acb98ba4f72563</li>
</ul></details>

<details><summary>CTS tests (Failed: 0/228032)</summary><ul><li>Built with version 1.2.5.2</li>
<details open><summary>Rhel 8.2, Gfx10</summary><ul>

<li>Passed: 32108/57008 (56.3%)</li>
<li>Failed: 0/57008 (0.0%)</li>
<li>Not Supported: 24900/57008 (43.7%)</li>
</ul></details>

<details open><summary>Ubuntu 18.04, Gfx9</summary><ul>

<li>Passed: 32108/57008 (56.3%)</li>
<li>Failed: 0/57008 (0.0%)</li>
<li>Not Supported: 24900/57008 (43.7%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx8</summary><ul>

<li>Passed: 32488/57008 (57.0%)</li>
<li>Failed: 0/57008 (0.0%)</li>
<li>Not Supported: 24520/57008 (43.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx103</summary><ul>

<li>Passed: 35419/57008 (62.1%)</li>
<li>Failed: 0/57008 (0.0%)</li>
<li>Not Supported: 21589/57008 (37.9%)</li>
</ul></details>


</ul></details>

**Review by ruiling — APPROVED** [review]
LGTM, Thanks!

File: lgc/state/PalMetadata.cpp [hunk 1]
```diff
@@ -842,7 +842,7 @@ llvm::Type *PalMetadata::getLlvmType(StringRef tyName) const {
 // =====================================================================================================================
 // Updates the SPI_SHADER_COL_FORMAT entry.
 //
-void PalMetadata::updateSpiShaderColFormat(ArrayRef<ColorExportInfo> exps, bool hasDepthExpFmtZero, bool killEnabled) {
+void PalMetadata::updateSpiShaderColFormat(ArrayRef<ColorExportInfo> exps, bool hasDepthExpFmtZero) {
```
Comment (amdrexu) [inline lgc/state/PalMetadata.cpp:845]: Can you add the parameter comments? They are missing. It is not your fault, a old problem caused by others.

**Review by amdrexu — COMMENTED** [review]
LGTM

File: lgc/state/PalMetadata.cpp (hunk 1 above)
Comment (perlfu) [reply lgc/state/PalMetadata.cpp:847]: Done.

**Comment by amdvlk-admin** [comment]
### Test summary for commit 7ad00dc44ded1e1e12dd9df75ebd4250ebb43fdf
<details><summary>Driver commits used in build</summary><ul>

<li>CWPACK: amd-master 7387247eb9889ddcabbc1053b9c2052e253b088e</li>
<li>METROHASH: amd-master 3c566dd9cda44ca7fd97659e0b53ac953f9037d2</li>
<li>PAL: dev 408fbcdd1249dea3469bbd3426c29ca344386d6f</li>
<li>SPVGEN: dev bb3da8bc00bcf4373d563bf9b3c74a80a8b28fe6</li>
<li>XGL: dev 5cb0bf59dd19a17fc8a6aba2743e6dbdba675195</li>
<li>LLVM-PROJECT: amd-gfx-gpuopen-dev 35321a037d4e75d01fbf6b2264acb98ba4f72563</li>
</ul></details>

<details><summary>CTS tests (Failed: 0/228032)</summary><ul><li>Built with version 1.2.5.2</li>
<details open><summary>Rhel 8.2, Gfx10</summary><ul>

<li>Passed: 32108/57008 (56.3%)</li>
<li>Failed: 0/57008 (0.0%)</li>
<li>Not Supported: 24900/57008 (43.7%)</li>
</ul></details>

<details open><summary>Ubuntu 18.04, Gfx9</summary><ul>

<li>Passed: 32108/57008 (56.3%)</li>
<li>Failed: 0/57008 (0.0%)</li>
<li>Not Supported: 24900/57008 (43.7%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx8</summary><ul>

<li>Passed: 32488/57008 (57.0%)</li>
<li>Failed: 0/57008 (0.0%)</li>
<li>Not Supported: 24520/57008 (43.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx103</summary><ul>

<li>Passed: 35419/57008 (62.1%)</li>
<li>Failed: 0/57008 (0.0%)</li>
<li>Not Supported: 21589/57008 (37.9%)</li>
</ul></details>


</ul></details>

**Review by amdrexu — APPROVED** [review]

**Pull request merged**